docs/src/ScriptingReference/core-callbacks.md

```diff
@@ -30,3 +31,24 @@ function on_script_unloaded()
     print("Goodbye world")
 end
 ```
+
+## `on_script_reloaded`
+
+This will be called twice: right before and after a script is reloaded.
+
+The first parameter `save` informs you whether it is time to save a value or restore it.
+
+Before the reload, it is called with one argument: `true`. After the script is reloaded, it is called with two parameters: the first is `false` and the second is value returned from before.
```

Comment:
as of now this seems to be misleading for non-shared contexts, in which case only the second half of the callback would happen

Reply:
ohh, I didn't catch the second conditional